### PR TITLE
[Bug Fix] Fix wrong value when choosing custom reminder date

### LINKED ISSUE #4248: The selection of a personalized date on the calendar reminder for a message doesn't works correctly.
### Steps to reproduce

1. Select the action "reminder for a message".
2. Set a personalized date and time (hour & minutes) to remember the message.  The date hasn't to be the same "weekday" than today is. The time hour isn't important.

### Expected behaviour

The personalized date and hour is correctly selected on the selector and so established.

### Actual behaviour

The personalized date is correctly selected on the selector, but when you accept it's not correctly establised/saved. The time(hour & minutes) is correctly selected.
The day selected is "changed automatically to something wrong".

Example:

Today (current date) is 19/09/2024 (Thursday).

Select a personalized remember date and time (hour & minutes) for a message, like 23/9/2024 at 16:48. 

The date and time that the selector shows is "26/09/2024 16:48" (Thursday).

### Device brand and model

S24

### Android version

14

### Nextcloud Talk app version

v20.0.1

### Nextcloud server version

28.0.10

### Talk version

v18.0.11

### Custom Signaling server configured

None

### Custom TURN server configured

None

### Custom STUN server configured

None

### Android logs

_No response_

### Server log

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by parneet-guraya**
fix #4248

## Before
https://github.com/user-attachments/assets/ead05132-3303-4dfb-b0bf-266dd7e510ec

## After
https://github.com/user-attachments/assets/a5a49119-ee9a-4035-961f-13b3fb4d4b79

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not needed
- [x] 🔖 Capability is checked or not needed 
- [ ] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [ ] 📅 Milestone is set
- [x] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

**Review by mahibi — APPROVED** [review]

**Comment by mahibi** [comment]
thank you @parneet-guraya :+1: 
Also waiting for a review from @rapterjet2004

**Comment by mahibi** [comment]
this also resolves https://github.com/nextcloud/talk-android/issues/4099

**Comment by parneet-guraya** [comment]
Nice 👍

**Review by rapterjet2004 — APPROVED** [review]

**Pull request merged**
